src/useFormState.js

```diff
@@ -256,16 +257,7 @@ export default function useFormState(initialState, options) {
         inputOptions.onBlur(e);
         formOptions.onBlur(e);
 
-        /**
-         * Limiting input validation on blur to:
-         * A) when it's either touched for the first time
-         * B) when it's marked as dirty due to a value change
-         */
-        /* istanbul ignore else */
-        if (!formState.current.touched[name] || isDirty(name)) {
-          validate(e);
-          setDirty(name, false);
-        }
+        if (inputOptions.validateOnBlur) validate(e);
```

Comment:
> inputProps.onBlur was erroneously invoking validate when inputOption.validateOnBlur is false.

This is intentional behavior, and not a bug. Validation is intended to be executed on both blur and change events by default.

Validation needs to be called on blur by default in order to catch required fields if the user skipped them and without any typing anything. Here's an example: https://codesandbox.io/s/validationonblur-kqoj9?fontsize=12

The purpose of `validateOnBlur` is to _limit_ validation to the blur event only and prevent the `change` event from triggering validation.